plugins/modules/system/sudoers.py

```diff
@@ -158,11 +170,20 @@ def content(self):
         runas_str = '({runas})'.format(runas=self.runas) if self.runas is not None else ''
         return "{owner} ALL={runas}{nopasswd} {commands}\n".format(owner=owner, runas=runas_str, nopasswd=nopasswd_str, commands=commands_str)
 
+    def validate(self):
+        check_command = ['visudo', '-c', '-f', '-']
+        rc, stdout, stderr = self.module.run_command(check_command, data=self.content())
+
+        if rc != 0:
+            raise Exception('Failed to validate sudoers rule:\n{stdout}'.format(stdout=stdout))
+
     def run(self):
         if self.state == 'absent' and self.exists():
             self.delete()
             return True
 
+        self.validate()
```

Comment:
I'm wondering whether this shouldn't be something that should be requested by the user, instead of be running by default. Is it possible that files are invalid now that - without changing their content - will be valid later? I.e. does `visudo -c` check whether executables mention in the file actually exist?

Also is `-f -` supported by all versions, or does this introduce a new requirement?

Reply:
How about:
- Don't simply call `visudo`, but use `module.get_bin_path('visudo', required=False)` to get hold of its path first, and only check if it is not `None`, and emit a warning otherwise.
- Potentially have an option which allows to always disable it, or always enforce it. Default value should be something like auto-detect, which behaves as sketched above. (When enforced, it should fail if it cannot find `visudo`.)

I think it will be OK to add the option also in a later version if it doesn't fail when `visudo` isn't available, though it would be great if the option is around as well so folks can get rid of the warning if `sudo` isn't installed (yet) :)